ext/opentelemetry-exporter-cloud-trace/src/opentelemetry/exporter/cloud_trace/__init__.py

```diff
@@ -0,0 +1,284 @@
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id), cloud_trace_spans,
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.error("Error while writing to Cloud Trace", exc_info=ex)
+            return SpanExportResult.FAILURE
+
+        return SpanExportResult.SUCCESS
+
+    def _translate_to_cloud_trace(
+        self, spans: Sequence[Span]
+    ) -> List[Dict[str, Any]]:
+        """Translate the spans to Cloud Trace format.
+
+        Args:
+            spans: Tuple of spans to convert
+        """
+
+        cloud_trace_spans = []
+
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = _get_hexadecimal_trace_id(ctx.trace_id)
+            span_id = _get_hexadecimal_span_id(ctx.span_id)
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if span.parent:
+                parent_id = _get_hexadecimal_span_id(span.parent.span_id)
+
+            start_time = _get_time_from_ns(span.start_time)
+            end_time = _get_time_from_ns(span.end_time)
+
+            attributes = _extract_attributes(span.attributes)
+
+            cloud_trace_spans.append(
+                {
+                    "name": span_name,
+                    "span_id": span_id,
+                    "display_name": _get_truncatable_str_object(
+                        span.name, 128
+                    ),
+                    "start_time": start_time,
+                    "end_time": end_time,
+                    "parent_span_id": parent_id,
+                    "attributes": attributes,
+                    "links": _extract_links(span.links),
+                    "status": _extract_status(span.status),
+                    "time_events": _extract_events(span.events),
+                }
+            )
+            # TODO: Leverage more of the Cloud Trace API, e.g.
+            #  same_process_as_parent_span and child_span_count
+
+        return cloud_trace_spans
+
+    def shutdown(self):
+        pass
+
+
+def _get_hexadecimal_trace_id(trace_id: int) -> str:
+    return "{:032x}".format(trace_id)
+
+
+def _get_hexadecimal_span_id(span_id: int) -> str:
+    return "{:016x}".format(span_id)
+
+
+def _get_time_from_ns(nanoseconds: int) -> Dict:
+    """Given epoch nanoseconds, split into epoch milliseconds and remaining
+    nanoseconds"""
+    if not nanoseconds:
+        return None
+    seconds, nanos = divmod(nanoseconds, 1e9)
+    return {"seconds": int(seconds), "nanos": int(nanos)}
+
+
+def _get_truncatable_str_object(str_to_convert: str, max_length: int):
+    """Truncate the string if it exceeds the length limit and record the
+    truncated bytes count."""
+    truncated, truncated_byte_count = _truncate_str(str_to_convert, max_length)
```

Comment:
This looks like it's assuming one byte per character now?

```python
>>> _truncate_str("中文翻译", 4)
('中文翻译', 0)

>>> len("中文翻译".encode('utf-8'))
12
```

I think your first approach was right:

```
>>> "中文翻译".encode('utf-8')[:4].decode('utf-8', errors='ignore')
'中'
```

You'd just need to compare the re-encoded string length to get the actual truncated byte count.

Reply:
`str_to_convert` may well always be ASCII, but in that case the docs should say so.

Reply:
Ah good catch, I was making that assumption (incorrectly).